dask/optimization.py

```diff
@@ -345,17 +346,23 @@ def inline_functions(
         dependencies = {k: get_dependencies(dsk, k) for k in dsk}
     dependents = reverse_dict(dependencies)
 
-    def inlinable(v):
-        try:
-            return functions_of(v).issubset(fast_functions)
-        except TypeError:
-            return False
-
-    keys = [
-        k
-        for k, v in dsk.items()
-        if istask(v) and dependents[k] and k not in output and inlinable(v)
-    ]
+    def inlinable(key, task):
+        if (
+            not isinstance(task, BaseTask)
```

Comment:
Are `Tasks` never inlineable? Is this because we inline elsewhere?

Reply:
this is because legacy and new don't play well with each other. 

The `convert_legacy_*` can handle a mix of tasks but other layers cannot (so far, only distributed-only graphs come with `Task` objects to the conversion in distributed alone is sufficient. The very next step here should be to do a similar change in dask/dask to allow internals to assume we're working in the new system.


TLDR for now, legacy low level optimization is/should be disabled whenever a BaseTask is encountered.